code/GroundStation.ino

```diff
@@ -0,0 +1,293 @@
+/* JEZDZCY JARZYNOWEJ - CAN SAT 2016/17
+ * GROUND STATION RECEIVER 
+ * CONNECTIONS:
+ * A4 - barometer SCA
+ * A5 - barometrer SCL
+ * 2 - radio TX
+ * 3 - radio RX
+ * 10 - micrSD CS
+ * 11 - microSD MOSI
+ * 12 - microSD MISO
+ * 13 - microSd SCLK
+ */
+#include <LPS331.h>
+#include <SPI.h>
+#include <SD.h>
+#include <Wire.h>
+#include <SoftwareSerial.h>
+
+SoftwareSerial radioSerial(2, 3);
+
+int startTime;
+const int serialDelay = 1000;
+const String autoSaveTime = "60";
+
+class TheTime{
+  public:
+    int inHour;
+    int inMinute;
+    int inSecond;
+    int inMilisecond;
+    
+    int hour;
+    int minute;
+    int second;
+    unsigned long milisecond;
+    
+    void calculate(){
+      while(milisecond >= 1000){
+        second++;
+        milisecond -= 1000;
+      }
+      while(second >= 60){
+        minute++;
+        second -= 60;
+      }
+      while(minute >= 60){
+        hour++;
+        second -= 60;
+      }
+    }
+
+    void clearTimes(){
+      hour = inHour;
+      minute = inMinute;
+      second = inSecond;
+      milisecond = inMilisecond;
+    }
+  
+    void calculateTime(){
+      clearTimes();
+      milisecond = inMilisecond + millis() - startTime;
+      calculate();
+    }
+};
+
+const int chipSelect = 10;
+File usedFile;
+
+LPS331 sensor;
+
+TheTime theTime;
+
+String sdError = "SDERROR";
```

Comment:
To można zrobić jako dyrektywy preprocesora, wyjdzie taniej pamięciowo. W sensie #define SDERROR "SDERROR" czy coś podobnego